docs/database/azure-cosmos-db-integration.md

```diff
@@ -87,6 +89,142 @@ public class ExampleService(
+
+var app = builder.Build();
+
+app.MapGet("/api/customers", async (
+    [FromKeyedServices("customers")] Database database) =>
+{
+    // Get container from database and query data
+});
+
+app.MapPost("/api/orders", async (
+    [FromKeyedServices("orders")] Database database,
+    [FromBody] OrderRequest order) =>
+{
+    // Get container from database and query data
+});
+
+app.Run();
+```
+
+The preceding example code demonstrates how to register two databases, `details` and `customers`. Each named database can be used to get their corresponding containers to query data.
+
+### Add Azure Cosmos DB container
+
+<!-- TODO: Add xref to AddAzureCosmosContainer when available -->
+
+When you add a Cosmos DB resource in the app host project, you can also add an Azure Cosmos DB container resource as well. The container resource is considered a child resource to the parent <xref:Aspire.Hosting.AzureCosmosDBResource>. In your client-consuming project, you can deep-link to the container resource by name, registering a <xref:Microsoft.Azure.Cosmos.Container> instance for use with dependency injection. For example, consider the following code that calls `AddAzureCosmosContainer` on an <xref:Microsoft.Extensions.Hosting.IHostApplicationBuilder> instance:
+
+```csharp
+builder.AddAzureCosmosContainer(connectionName: "details");
+```
+
+You can then retrieve the `Container` instance using dependency injection. For example, to retrieve the container from a delegate in a <xref:Microsoft.AspNetCore.Builder.EndpointRouteBuilderExtensions.MapGet*> call consider the following code:
+
+```csharp
+app.MapGet("/api/orders/{id:guid}", async (
+    Container container, 
+    [FromRoute] Guid id) =>
+{
+    // Query data from container...
+});
+```
+
+### Add keyed Azure Cosmos DB container
+
+<!-- TODO: Add xref to AddKeyedAzureCosmosContainer when available -->
+
+There's also an `AddKeyedAzureCosmosContainer` method that allows you to register multiple containers with different connection names. For example, consider the following code that calls `AddKeyedAzureCosmosContainer` on an <xref:Microsoft.Extensions.Hosting.IHostApplicationBuilder> instance:
+
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+
+builder.AddKeyedAzureCosmosContainer("customers");
+
+var app = builder.Build();
+
+app.MapGet("/api/customers", async (
+    [FromKeyedServices("customers")] Container container) =>
+{
+    // Query data from container...
+});
+
+app.Run();
+```
+
+If you have multiple containers under the same database connection, you can use the `AddAzureCosmosDatabase` API to attach multiple containers under the same database connection. All child containers share the same <xref:Azure.Cosmos.CosmosClient> and database connection. This strategy is useful when associating the same <xref:Azure.Cosmos.CosmosClientOptions> with multiple containers. Consider the following alternative code, to register multiple containers under the same database connection:
```

Comment:
Should we move this section earlier to where we talk about `AddCosmosDatabase`. Right now, we allude to the multiple container thing in the database section but don't get back to talking about it under here.

Reply:
Those are separate sections, hosting vs client integrations, no?